django/db/models/expressions.py

```diff
@@ -689,6 +689,17 @@ def as_sql(self, compiler, connection):
     def get_group_by_cols(self, alias=None):
         return [self]
 
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None,
+                           summarize=False, for_save=False, simple_col=False):
+        other_fields = (set(query.model._meta.get_fields()) -
+                        set(query.model._meta.get_fields(include_parents=False)))
+        for field in other_fields:
+            if field.name in self.sql:
```

Comment:
We should use `column` from `get_attname_column()` instead of `name` because fields may have a custom `db_column`. I would also use only `local_fields` from parents. Moreover current `break` doesn't work if `RawSQL` contains field from different parents.

Maybe sth like: 
```python
def resolve_expression(self, query=None, allow_joins=True, reuse=None,
                       summarize=False, for_save=False, simple_col=False):
    # Resolve parents fields used in raw SQL.
    for parent in query.model._meta.get_parent_list():
        for parent_field in parent._meta.local_fields:
            _, column_name = parent_field.get_attname_column()
            if column_name.lower() in self.sql.lower():
                query.resolve_ref(parent_field.name, allow_joins, reuse, summarize, simple_col)
                break
    return super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
```